src/main/java/tempest/GUI/AddSessionPage.java

```diff
@@ -0,0 +1,184 @@
+
+//TODO:
+// - Add a check so the user can't enter a session > 24hrs
+// - Add a check so the user can't enter study sessions in a day which add up to be > 24hrs
+// - Better if pressing enter sends user back a screen? - Feels more like you've entered something
+
+public class AddSessionPage extends Page implements ActionListener{
+    private final GUIManager manager;
+    private final State state;
+    private final GUIComponents components = new GUIComponents();
+
+    private JComboBox<Object> moduleDropDown;
+    private JTextField hoursInput;
+    private JTextField minutesInput;
+    private JButton enterButton;
+
+    public AddSessionPage(State state, GUIManager guiManager){
+        this.state = state;
+        this.manager = guiManager;
+    }
+
+    public JPanel getPanel(){
+        JPanel sessionPanel = new JPanel();
+
+        JPanel buttonPanel = components.getButtonPanel(manager, this);
+        enterButton = (JButton) buttonPanel.getComponent(1);
+
+        JPanel inputPanel = new JPanel();
+
+        moduleDropDown = components.getModuleDropDown();
+        inputPanel.add(moduleDropDown);
+
+        hoursInput = new JTextField(2);
+        JLabel hoursInputLabel = new JLabel("Hours");
+
+        minutesInput = new JTextField(2);
+        JLabel minutesInputLabel = new JLabel("Minutes");
+
+        JPanel hoursPanel = new JPanel();
+        hoursPanel.add(hoursInputLabel);
+        hoursPanel.add(hoursInput);
+
+        JPanel minutesPanel = new JPanel();
+        minutesPanel.add(minutesInputLabel);
+        minutesPanel.add(minutesInput);
+
+        JPanel timeInputPanel = new JPanel();
+        timeInputPanel.add(hoursPanel);
+        timeInputPanel.add(minutesPanel);
+        timeInputPanel.setLayout(new FlowLayout());
+
+        inputPanel.add(timeInputPanel);
+
+        sessionPanel.add(inputPanel);
+        sessionPanel.add(buttonPanel);
+
+        sessionPanel.setLayout(new BoxLayout(sessionPanel, BoxLayout.Y_AXIS));
+
+        return sessionPanel;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e){
+        Object source = e.getSource();
+
+        if(source == enterButton){
+            handleAddingSession();
+        }
+    }
+
+    /**
+     * Handles the user trying to enter a session
+     */
+    private void handleAddingSession() {
+        if (moduleDropDown.getItemCount() < 1) { // No modules created, so can't add a session
+            return;
+        }
+
+        String moduleName = Objects.requireNonNull(moduleDropDown.getSelectedItem()).toString();
+
+        Date date = new Date();
+
+        String hours = hoursInput.getText();
+        String minutes = minutesInput.getText();
+
+        int hoursInt = 0;
+        int minutesInt = 0;
+
+        hours = hours.strip();
+        minutes = minutes.strip();
+
+        if (hours.equals("") && minutes.equals("")) { // Empty hours and minutes
+            System.out.println("A session needs to be longer >= 1 minute");
```

Comment:
For warnings like these, should we have a task that displays alerts to the user within the GUI? Not saying it needs to be part of this task, but do you think it's a good idea?

Reply:
Do you mean as a separate error box GUI window or within the current window?

Reply:
I was going to say it wouldn't worth the time & effort to set this up. But, I've just found this:

https://docs.oracle.com/javase/tutorial/uiswing/components/dialog.html

So I'll implement this now.